### ISSUE TITLE
Crash when running invalid query with extra semicolon

### ISSUE THREAD

**Issue by samuelbradshaw** [Sequel-Ace/Sequel-Ace#791]
- Sequel Ace Version (including build number): 3.1.0 (3013)
- Sequel Ace Source (App Store/GitHub/Homebrew): App Store
- macOS Version: 10.15.7 (Catalina)
- Processor Type (Intel/Apple): Intel
- MySQL Version: 8.0.22
- macOS Localization: English (US)

**Is Issue Present in [Latest Beta](https://github.com/Sequel-Ace/Sequel-Ace/releases)?**
N/A (no current beta)

**Description**
I accidentally ran a query with the semicolon in the wrong place, and the app crashed:
```
SELECT id, title FROM content_song;
WHERE title LIKE '%morning%'
```

**Steps To Reproduce**
1. Paste the query above (or something similar) into the query editor.
2. Run the query using the "Run All" option (this is what I have set as default).

**Expected Behaviour**
The app should show an error message when attempting to run the query.

**Actual Behaviour**
The app crashes.

**Related Issues**
The query doesn't show in Query History or the Sequel Ace console after the app restarts, so it may be similar to the crash in issue #773.

**Additional Context**
Crash log attached.

[Sequel Ace_2021-01-14-163528_Samuel-Bradshaw-MacBook-Pro.crash.zip](https://github.com/Sequel-Ace/Sequel-Ace/files/5817527/Sequel.Ace_2021-01-14-163528_Samuel-Bradshaw-MacBook-Pro.crash.zip)

**Comment by Jason-Morcos** [comment]
Crashes for me too! I love a 100% reproducible bug! :)

**Comment by jamesstout** [comment]
I think I fixed this recently ... just checking.

**Comment by jamesstout** [comment]
Can't reproduce. Where is your cursor in the text window?

**Comment by jamesstout** [comment]
ah, this is the line:

https://github.com/Sequel-Ace/Sequel-Ace/blob/5f2e52cc2c11d62631eeee9238f9487e0e7e1502/Source/Controllers/MainViewControllers/SPCustomQuery.m#L851

And I fixed in 4ad193f2. Merged but not released yet.

**Comment by jamesstout** [comment]
@Sequel-Ace/all What do we think about highlighting the error in red? (Obviously the text there is just test, it wouldn't be red if there were no errors!)

![Screenshot 2021-01-15 at 4 46 32 PM](https://user-images.githubusercontent.com/1576190/104705712-394b3980-5755-11eb-9bf2-d975c01195c8.png)

Looks OK for the various kinds of red/green colour-blindness:

![Screenshot 2021-01-15 at 4 46 38 PM](https://user-images.githubusercontent.com/1576190/104705943-77485d80-5755-11eb-8918-ca6dc01e5792.png)
![Screenshot 2021-01-15 at 4 46 47 PM](https://user-images.githubusercontent.com/1576190/104705949-79aab780-5755-11eb-8332-e4223994fdb6.png)
![Screenshot 2021-01-15 at 4 46 42 PM](https://user-images.githubusercontent.com/1576190/104705951-7a434e00-5755-11eb-94a6-5605b46a0cb6.png)
![Screenshot 2021-01-15 at 4 46 51 PM](https://user-images.githubusercontent.com/1576190/104705954-7adbe480-5755-11eb-98f7-ebc5a4fa7ea5.png)

**Comment by Jason-Morcos** [comment]
> @Sequel-Ace/all What do we think about highlighting the error in red? (Obviously the text there is just test, it wouldn't be red if there were no errors!)
> 
> ![Screenshot 2021-01-15 at 4 46 32 PM](https://user-images.githubusercontent.com/1576190/104705712-394b3980-5755-11eb-9bf2-d975c01195c8.png)
> 
> Looks OK for the various kinds of red/green colour-blindness:
> 
> ![Screenshot 2021-01-15 at 4 46 38 PM](https://user-images.githubusercontent.com/1576190/104705943-77485d80-5755-11eb-8918-ca6dc01e5792.png)
> ![Screenshot 2021-01-15 at 4 46 47 PM](https://user-images.githubusercontent.com/1576190/104705949-79aab780-5755-11eb-8332-e4223994fdb6.png)
> ![Screenshot 2021-01-15 at 4 46 42 PM](https://user-images.githubusercontent.com/1576190/104705951-7a434e00-5755-11eb-94a6-5605b46a0cb6.png)
> ![Screenshot 2021-01-15 at 4 46 51 PM](https://user-images.githubusercontent.com/1576190/104705954-7adbe480-5755-11eb-98f7-ebc5a4fa7ea5.png)

I like that idea, @jamesstout! Nice little visual clue!

**Comment by Jason-Morcos** [comment]
> ah, this is the line:
> 
> https://github.com/Sequel-Ace/Sequel-Ace/blob/5f2e52cc2c11d62631eeee9238f9487e0e7e1502/Source/Controllers/MainViewControllers/SPCustomQuery.m#L851
> 
> And I fixed in [4ad193f](https://github.com/Sequel-Ace/Sequel-Ace/commit/4ad193f290ec9641e028b96e3ee187d844b70083). Merged but not released yet.

Closing issue as fix will be included in next release!
